File: app/backend/src/routes/users/verify_session.rs
// src/routes/users/logout.rs
use crate::prelude::*;
use crate::types::ErrorResponse;

#[tracing::instrument(name = "Verify user session", skip(redis_pool, req))]
#[actix_web::get("/verify-session")]
pub async fn verify_session(
    req: HttpRequest,
    redis_pool: web::Data<deadpool_redis::Pool>
) -> HttpResponse {
    tracing::info!(target: "backend", "Verifying session");

    let sss_uuid_token =
        if let Some(sss_uuid_cookie) = req.cookie("session_uuid") {
            sss_uuid_cookie.value().to_string()
        } else {
            return HttpResponse::BadRequest().json(
                ErrorResponse { error: "Session cookie missing.".to_string() }
            );
        };

    match crate::utils::verify_session_token(sss_uuid_token, &redis_pool).await {
        Ok(_) =>
            HttpResponse::Ok().finish(),
        Err(e) =>
            HttpResponse::Unauthorized().json(
                ErrorResponse { error: format!("Failed to verify session: {}", e) }
            )
    }
}

File: app/backend/nexis-py/nexis_py/routes/users/verify_session.py
from fastapi import Cookie
from nexis_py.utils.auth.tokens import verify_session_token
from nexis_py.types.responses import ErrorResponse
from . import router

@router.get("/verify-session")
def verify_session(sss_uuid_token: str = Cookie(None, alias="session_uuid")):
    if not sss_uuid_token:
        return ErrorResponse("`session_uuid` cookie is not present.", 400)

    response = verify_session_token(sss_uuid_token)
    return response